cfme/configure/settings.py

```diff
@@ -23,109 +25,173 @@ class TimeProfileAddForm(View):
+            timezone: It's the required Time Zone for the time_profile.
+            cancel (bool) : It's a flag used to cancel or not the create operation.
+
+        return: It returns the object of the newly created time_profile object.
+        """
+
+        time_profile = self.instantiate(description, scope, days, hours, timezone)
 
-    def delete(self):
         view = navigate_to(self, 'All')
-        rows = view.timeprofile_form.table
-        for row in rows:
-            if row.description.text == self.description:
-                row[0].check()
-        view.timeprofile_form.configuration.item_select("Delete selected "
-                                                        "Time Profiles", handle_alert=True)
+        view.time_profile_form.configuration.item_select('Add a new Time Profile')
+        view.time_profile_form.fill({
+            'description': description,
+            'scope': scope,
+            'days': days,
+            'hours': hours,
+            'timezone': timezone,
+        })
+        if not cancel:
+            view.time_profile_form.save_button.click()
+            view.flash.assert_message('Time Profile "{}" was saved'.format(description))
+        return time_profile
+
+    def delete(self, cancel=False, *time_objs):
+        """
+        This method performs the delete operation.
+
+        Args:
+            cancel (bool) : It's a flag used for selecting Ok or Cancel from delete confirmation
+             dialogue box
+            time_objs : It's time profile object.
+        """
+
+        view = navigate_to(self, 'All')
+        rows = view.time_profile_form.table
+        for time_obj in time_objs:
+            for row in rows:
+                if row.description.text == time_obj.description:
+                    row[0].check()
+        if not cancel:
+            view.time_profile_form.configuration.item_select("Delete selected Time Profiles",
+                                                             handle_alert=True)
+        else:
+            view.time_profile_form.configuration.item_select("Delete selected Time Profiles",
+                                                             handle_alert=False)
+        view.flash.assert_no_error()
 
 
-@navigator.register(Timeprofile, 'All')
-class TimeprofileAll(CFMENavigateStep):
+@navigator.register(TimeProfileCollection, 'All')
+class TimeProfileCollectionAll(CFMENavigateStep):
     VIEW = TimeProfileAddFormView
     prerequisite = NavigateToAttribute('appliance.server', 'MySettings')
 
     def step(self):
         self.view.mysetting.tabs.time_profile.select()
 
 
+@navigator.register(TimeProfile, 'Edit')
+class TimeProfileEdit(CFMENavigateStep):
+    VIEW = TimeProfileEditView
+    prerequisite = NavigateToAttribute('parent', 'All')
+
+    def step(self):
+        self.view.time_profile_form.configuration.item_select('Edit selected Time Profile')
```

Comment:
**Required**

How has the time profile been selected in the view? The prerequisite destination has only navigated to the collection of Time Profiles.